doc/guide/sample.hpp

```diff
@@ -16,6 +16,94 @@ e.g., @c make @c mlpack_knn or @c make @c mlpack_test; see @ref build.  However,
 any program that uses mlpack (and is not a part of the library itself) can be
 compiled easily with g++ or clang from the command line.
 
+@section compile_link_and_run Compile, Link and Run C++ Files 
+
+Typically the following commands would be used to compile and run the C++ files.
+
+@subsection compile_using_gcc Compile Using GCC
+
+@code{.sh}
+g++ -std=c++11 filename.cpp -o filename -lmlpack -larmadillo
+./filename
+@endcode
+
+If @c pkg-config is installed:
+
+@code{.sh}
+g++ -std=c++11 filename.cpp -o filename \
+`pkg-config --cflags --libs mlpack armadillo`
+./filename
+@endcode
+
+@subsection compile_using_clang Compile Using Clang
+
+@code{.sh}
+clang++ -Wall filename.cpp -o filename -lmlpack -larmadillo
+./filename
+@endcode
+
+If @c pkg-config is installed:
+
+@code{.sh}
+clang++ -Wall filename.cpp -o filename \
+`pkg-config --cflags --libs mlpack armadillo`
+./filename
+@endcode
+
+@note
+Flags like @c -lboost_serialization might be required depending upon the 
+code. If you are using LAPACK and BLAS instead of Armadillo the @c -llapack
+and @c -lblas would be required instead of @c -larmadillo
```

Comment:
Armadillo itself uses LAPACK and BLAS, so the right way to put this would be "if you have Armadillo configured with `ARMA_USE_WRAPPER` not set in `armadillo_bits/config.hpp`, you will need to link directly against LAPACK and BLAS with @c -llapack and @c -lblas instead of @c -larmadillo."